Plutarch/NonZero.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Plutarch.NonZero (PNonZero, pnonZero, ptryNonZero) where
```

Comment:
I assume the current convention to enforce invariants for types is to make sure the constructor is not exported? You can't have manual invariant checks in `PlutusType` anymore.

If so, `PValue`'s constructor needs to be unexported.

Reply:
yep